app/src/app/components/contents/explorer/Table.tsx

```diff
@@ -10,14 +10,22 @@ const headers: Header[] = [
 ];
 
 export default function Table({data}: PacketTableProps) {
-    const [ascending, setAscending] = useState(true);
+    const [ascendingName, setAscendingName] = useState(false);
+    const [ascendingStatus, setAscendingStatus] = useState(false);
 
     const sortPackets = (accessor: string) => {
-        data.sort((a, b) => {
-            setAscending(!ascending);
-            return (`${a[accessor as keyof Packet]}`
-                .localeCompare(`${b[accessor as keyof Packet]}`)) * (ascending ? -1 : 1);
-        });
+        let ascending = false;
+        if (accessor === "displayName") {
+            ascending = ascendingName;
+            setAscendingName(!ascending);
+            setAscendingStatus(false);
+        } else if (accessor === "published") {
+            ascending = ascendingStatus;
+            setAscendingStatus(!ascending);
+            setAscendingName(false);
+        }
+        data.sort((a, b) => (`${a[accessor as keyof Packet]}`
+            .localeCompare(`${b[accessor as keyof Packet]}`)) * (ascending ? -1 : 1));
```

Comment:
```suggestion
        const newAscending = accessor === ascendingCol ? !ascending : true;
        setAscending(newAscending);
        setAscendingCol(accessor);
        
        data.sort((a, b) => (`${a[accessor as keyof Packet]}`
            .localeCompare(`${b[accessor as keyof Packet]}`)) * (newAscending ? -1 : 1));
```

Reply:
Sorry, I should have called it `sortCol'`rather than `ascendingCol`, I just meant the column that you're sorting by (whether asc or desc), and `ascending` is still the flag to say whether it's currently ascending or descending. 
So if you've got sortCol="displayName" and ascending=false, then you click on "status", it should then set `newAscending` to true since accessor ("status") != sortCol ("displayName"), so it should then sort by status ascending.

Reply:
OK, I think I just had the sort direction the wrong way round. This works for me:
```
    const [ascending, setAscending] = useState(true);
    const [sortCol, setSortCol] = useState("");

    const sortPackets = (accessor: string) => {
        const newAscending = (accessor === sortCol) ? !ascending : true;
        setAscending(newAscending);
        setSortCol(accessor);

        data.sort((a, b) => (`${a[accessor as keyof Packet]}`
            .localeCompare(`${b[accessor as keyof Packet]}`)) * (newAscending ? 1 : -1));
    };
```